src/overlays/actors/ovl_En_Tk/z_en_tk.c

```diff
@@ -17,62 +51,938 @@ const ActorInit En_Tk_InitVars =
+    point += this->currentWaypoint;
+
+    dx = point->x - this->actor.posRot.pos.x;
+    dz = point->z - this->actor.posRot.pos.z;
+
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y,
+                            func_800FD250(dx, dz) * (0x8000 / M_PI),
+                            10, 1000, 1);
+    this->actor.posRot.rot = this->actor.shape.rot;
+
+    if (SQ(dx) + SQ(dz) < 10.f)
+    {
+        this->currentWaypoint++;
+        if (this->currentWaypoint >= path->nPoints)
+        {
+            this->currentWaypoint = 0;
+        }
+
+        return 0;
+    }
+    else
+    {
+        return 1;
+    }
+}
+
+u16 func_80B1C54C(GlobalContext *ctxt, Actor *a1)
+{
+    u16 ret;
+
+    ret = func_8006C360(ctxt, 14);
+    if (ret != 0)
+    {
+        return ret;
+    }
+
+    if (gSaveContext.inf_table[13] & 0x0200)
+    {
+        /* "Do you want me to dig here? ..." */
+        return 0x5019;
+    }
+    else
+    {
+        /* "Hey kid! ..." */
+        return 0x5018;
+    }
+}
+
+s16 func_80B1C5A0(GlobalContext *ctxt, Actor *actor)
+{
+    s32 ret = 1;
+
+    switch (func_8010BDBC(&ctxt->msgCtx))
+    {
+        case 0:
+        {
+            break;
+        }
+        case 1:
+        {
+            break;
+        }
+        case 2:
+        {
+            /* "I am the boss of the carpenters ..." (wtf?) */
+            if (actor->textId == 0x5028)
+            {
+                gSaveContext.inf_table[13] |= 0x0100;
+            }
+            ret = 0;
+            break;
+        }
+        case 3:
+        {
+            break;
+        }
+        case 4:
+        {
+            if (func_80106BC8(ctxt) != 0 &&
+                (actor->textId == 0x5018 || actor->textId == 0x5019))
+            {
+                if (ctxt->msgCtx.choiceIndex == 1)
+                {
+                    /* "Thanks a lot!" */
+                    actor->textId = 0x0084;
+                }
+                else if (gSaveContext.rupees < 10)
+                {
+                    /* "You don't have enough Rupees!" */
+                    actor->textId = 0x0085;
+                }
+                else
+                {
+                    ctxt->msgCtx.msgMode = 0x37;
+                    Rupees_ChangeBy(-10);
+                    gSaveContext.inf_table[13] |= 0x0200;
+                    return 2;
+                }
+                func_8010B720(ctxt, actor->textId);
+                gSaveContext.inf_table[13] |= 0x0200;
+            }
+            break;
+        }
+        case 5:
+        {
+            if (func_80106BC8(ctxt) != 0 &&
+                (actor->textId == 0x0084 || actor->textId == 0x0085))
+            {
+                func_80106CCC(ctxt);
+                ret = 0;
+            }
+            break;
+        }
+        case 6:
+        {
+            break;
+        }
+        case 7:
+        {
+            break;
+        }
+        case 8:
+        {
+            break;
+        }
+        case 9:
+        {
+            break;
```

Comment:
Do all those empty cases need to be separate? Otherwise I suggest grouping them together and/or omitting those that are unnecessary.

Reply:
The jump table has 10 cases and an offset of 0, so 0 and 9 were present in the original source code and are necessary to match. Looking at `func_8010BDBC`, it can return all values from 0 to 9, which hints that the return value is an enum. Omitting all cases except 0, 2, 4, 5, and 9 matches, but it seems plausible to me that the original code included a case for all enum values for completeness (I can't think of a reason why you'd have a redundant case for only the first and last value). There are switch statements of the return value from `func_8010BDBC` whose layout are identical to this elsewhere in the repo, which supports this hypothesis. For this reason I would rather not omit them, but I will remove the redundant block statements for the sake of brevity, if that's acceptable.